.github/actions/checkout-go/action.yml

```diff
@@ -0,0 +1,62 @@
+name: 'Checkout a repository for a Go project'
+description: 'Checkout and setup the environment for a Go project'
+inputs:
+  repository:
+    description: 'Repository name with owner.'
+    required: false
+    # Same default as https://github.com/actions/checkout/blob/main/action.yml#L6.
+    default: ${{ github.repository }}
+  ref:
+    # Note: the logic is fairly involved https://github.com/actions/checkout/blob/main/src/ref-helper.ts, 
+    # so we do not attempt to resolve it ourselves or provide a default value. We let the official `actions/checkout`
+    # do it for us.
+    description: 'The branch, tag or SHA to checkout.'
+    required: false
+  token:
+    description: 'The token to use.'
+    required: false
+    # Same default as https://github.com/actions/checkout/blob/main/action.yml#L24.
+    default: ${{ github.token }}
+  go-version:
+    description: 'The Go version to use, as expected by https://github.com/actions/setup-go.'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    # Note: we could use a single block:
+    #  `uses: actions/checkout
+    #     with:
+    #       ref: "${{ inputs.ref }}"`
+    # and it would work, because the ref field does not have a default
+    # value set https://github.com/actions/checkout/blob/main/action.yml#L7-L11.
+    # However, if this were to change in the future, we'd be setting an empty value
+    # when the developer has not defined it; and it would overwrite the default value
+    # set by the `actions/checkout`. Even if it is highlely unlikely the `actions/checkout` team
+    # will set a default value in the future, we want to be sure it does not affect us if they do.
+    # This is why we use 2 blocks to call the `actions/checkout`:
+    #   1. if inputs.ref != ''
+    #   2. if inputs.ref == ''
+    - name: Checkout the repository with user ref
+      if: inputs.ref != ''
+      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v2.3.4
+      with:
+        fetch-depth: 1
```

Comment:
I know we've always done this but do we really need to set fetch-depth? 1 is the default AFAICT

Reply:
you're right, let me remove it.